docusaurus/docs/Flutter/guides/adding_push_notifications_v2.mdx

```diff
@@ -0,0 +1,301 @@
+
+### Introduction
+
+Push notifications are a core part of the experience for a messaging app. Users often need to be notified
+of new messages and old notifications sometimes need to be updated silently.
+
+This guide details how to add push notifications to your app.
+
+You can read more about Stream’s [push delivery logic](https://getstream.io/chat/docs/flutter-dart/push_introduction/?language=dart#push-delivery-rules).
+
+### Setup FCM
+
+To integrate push notifications in your Flutter app, you need to use the package [firebase_messaging](https://pub.dev/packages/firebase_messaging).
+
+
+Follow the [Firebase documentation](https://firebase.flutter.dev/docs/messaging/overview/) to set up the plugin for Android and iOS.
+
+
+Once that's done, FCM should be able to send push notifications to your devices.
+
+### Integration With Stream
+
+#### Step 1 - Get the Firebase Credentials
+
+These credentials are the [private key file](https://firebase.google.com/docs/admin/setup#:~:text=To%20generate%20a%20private%20key%20file%20for%20your%20service%20account%3A) for your service account, in firebase console. 
+
+To generate a private key file for your service account, in the Firebase console:
+
+- Open Settings > Service Accounts.
+
+- Click **Generate New Private Key**, then confirm by clicking **Generate Key**.
+
+- Securely store the JSON file containing the key.
+
+This JSON file contains the credentials which needs to be uploaded to Stream’s server as explained in next step.
+
+#### Step 2 - Upload the Firebase Credentials to Stream
+
+You can upload your Firebase credentials using either the dashboard or the app settings API (available only in backend SDKs).
+
+##### Using the Stream Dashboard
+
+1. Go to the **Chat Overview** page on Stream Dashboard
+
+![](../assets/chat_overview_page-2fbd5bbfb70c5623bd37ff7d6c41bf4d.png)
+
+2. Enable **Firebase Notification** toggle on **Chat Overview**
+
+![](firebase_notifications_toggle-5aeabfcbdc24cb8f1fea7d41d0e845fc.png)
+
+3. Enter your Firebase Credentials and press "Save".
+
+##### Using the API
+
+You can also enable Firebase notifications and upload the Firebase credentials using one of our server SDKs.
+
+For example, using the JavaScript SDK:
+
+```js
+const client = StreamChat.getInstance('api_key', 'api_secret');
+client.updateAppSettings({
+  push_config: {
+    version: 'v2'
+  },
+  firebase_config: {
+    credentials_json: fs.readFileSync(
+      './firebase-credentials.json',
+      'utf-8',
+    ),
+ });
+```
+### Registering a Device With Stream Backend
+
+Once you configure a Firebase server key and set it up on Stream dashboard then a device that is supposed to receive push notifications needs to be registered on the Stream backend. This is usually done by listening for Firebase device token updates and passing them to the backend as follows:
+
+```dart
+firebaseMessaging.onTokenRefresh.listen((token) { 
+      client.addDevice(token, PushProvider.firebase); 
```

Comment:
when are we going to add multi-bundle support into this call ? a name to specify the type of saved firebase config

Reply:
can you send me a link to the documentation? I don't see that in the website

Reply:
ok found it, we'll add the parameter in another ticket